File: WorldSimulatorCPP/Grass.cpp
#include "Grass.h"
#include "World.h"
Grass::Grass(World* w, Coordinate c) {
	strength = 0;
	initiative = 0;
	species = "Grass";
	sprite = 'g';
	age = 0;
	world = w;
	coord = c;
	dead = false;
	w->addOrganism(this);
}

Grass::Grass(World* w, Coordinate c, int s, int a) {
	strength = s;
	initiative = 0;
	species = "Grass";
	sprite = 'g';
	age = a;
	world = w;
	coord = c;
	dead = false;
	w->addOrganism(this);
}

File: WorldSimulatorJAVA/Grass.java
package worldsimulator;

public class Grass extends Plant {
	
	public Grass(World w, Coordinate c) {
		strength = Species.GRASS.getStrength();
        initiative = Species.GRASS.getInitiative();
        species = Species.GRASS.getName();
        sprite = Species.GRASS.getSprite();
        age = 0;
        world = w;
        coord = c;
        dead = false;
        world.addOrganism(this);
    }

    public Grass(World w, Coordinate c, int s, int a) {
    	strength = s;
    	initiative = Species.GRASS.getInitiative();
        species = Species.GRASS.getName();
        sprite = Species.GRASS.getSprite();
        age = a;
        world = w;
        coord = c;
        dead = false;
        world.addOrganism(this);
    }
}
